webapp/patches/enketo-core+7.2.5.patch

```diff
@@ -14,20 +14,24 @@ index 62003af..db02cba 100644
               * Copied from relevant.js:
               *
 diff --git a/node_modules/enketo-core/src/js/relevant.js b/node_modules/enketo-core/src/js/relevant.js
-index 463af2c..f7dbe4d 100644
+index 463af2c..df8d235 100644
 --- a/node_modules/enketo-core/src/js/relevant.js
 +++ b/node_modules/enketo-core/src/js/relevant.js
-@@ -458,7 +458,9 @@ export default {
-                             : currentValue,
-                     });
-                 } else {
--                    if (isLeafNode && textContent !== '') {
-+                    // CHT PATCH
-+                    // /inputs is ALWAYS relevant #4875
-+                    if (isLeafNode && textContent !== '' && !path.endsWith('/inputs')) {
-                         node.textContent = '';
-                     }
-
+@@ -265,7 +265,13 @@ export default {
+                 alreadyCovered.push(node.getAttribute('name'));
+             }
+ 
+-            if (
++            // CHT-CORE PATCH
++            // /inputs is ALWAYS relevant #4875
++            if(!result && /^\/[^/]+\/inputs$/.test(path)) {
++                branchChange = this.enable(branchNode, path, options) || branchChange;
++                branchNode.classList.add('disabled');
++            }
++            else if (
```

Comment:
The previous patch here was attempting a much more surgical modification where the `inputs` group would actually still get set to be non-relevant, but we would not clear the data in the `inputs` group (as a part of [this logic](https://github.com/enketo/enketo-core/blob/9551bf19b42db3ee4637a2e7067b6e9ba62ced0e/src/js/relevant.js#L461). It would remain available to other expressions in the form.  Unfortunately, the side effect of this was that if any of the data in the non-relevant `inputs` changed, that change did not get stored/propagated to other expressions since technically the fields that changed were non-relevant.  So, when the db-object widget was triggered inside the non-relevant `inputs` group, it would load the data as expected, but it would not get saved into the form data model.  

This new patch takes a more heavy handed approach where we actually just never let the `inputs` group get set to be non-relevant (by hijacking [this logic flow](https://github.com/enketo/enketo-core/blob/9551bf19b42db3ee4637a2e7067b6e9ba62ced0e/src/js/relevant.js#L268)).  The hope is that this will let us avoid any other kinds of weird side effects at the data-model level (especially since Enekto has some additional logic that gets trippy when some fields _have never been relevant_). Basically this lets is side-step all of that and as far as the rest of the form is concerned the `inputs` group is always relevant.  The downside is that when the `inputs` relevant expression evaluates to `false`, _we do not want the user to actually see the `inputs` fields in the form.  So, that is why we still slap the `disabled` class onto the `inputs` group to make sure it is not shown in the form.